cookbook/recipes/markets/README.md

```diff
@@ -0,0 +1,200 @@
+
+
+> [!NOTE]
+> This recipe is particularly useful for existing Hydrogen projects. If you need to set up a brand new Hydrogen app, you can get a solid foundation by selecting the localization options when setting up your new project via the Shopify CLI. You can also use `h2 setup markets` to add localization support to your new Hydrogen app.
+
+## Requirements
+
+- Set up your store's regions and languages via [Shopify Markets](https://help.shopify.com/en/manual/markets).
+- Configure your products appropriately for each market.
+- Make sure your Hydrogen app is configured to use a default `language` and `country code`. They will be used as the fallback when no market is explicitly selected.
+
+
+## Ingredients
+
+_New files added to the template by this recipe._
+
+| File | Description |
+| --- | --- |
+| [app/components/CountrySelector.tsx](https://github.com/Shopify/hydrogen/blob/4fa2550c0cf7b07e2cf3b948b42c67cf3d552789/cookbook/recipes/markets/ingredients/templates/skeleton/app/components/CountrySelector.tsx) | A component that displays a country selector inside the Header. |
+| [app/components/Link.tsx](https://github.com/Shopify/hydrogen/blob/4fa2550c0cf7b07e2cf3b948b42c67cf3d552789/cookbook/recipes/markets/ingredients/templates/skeleton/app/components/Link.tsx) | A wrapper around the Remix Link component that uses the selected locale path prefix. |
+| [app/lib/i18n.ts](https://github.com/Shopify/hydrogen/blob/4fa2550c0cf7b07e2cf3b948b42c67cf3d552789/cookbook/recipes/markets/ingredients/templates/skeleton/app/lib/i18n.ts) | A helper function to get locale information from the context, a hook to retrieve the selected locale, and a list of available locales. |
+| [app/routes/($locale)._index.tsx](https://github.com/Shopify/hydrogen/blob/4fa2550c0cf7b07e2cf3b948b42c67cf3d552789/cookbook/recipes/markets/ingredients/templates/skeleton/app/routes/($locale)._index.tsx) | A route that renders a localized version of the home page. |
+| [app/routes/($locale).products.$handle.tsx](https://github.com/Shopify/hydrogen/blob/4fa2550c0cf7b07e2cf3b948b42c67cf3d552789/cookbook/recipes/markets/ingredients/templates/skeleton/app/routes/($locale).products.$handle.tsx) | A route that renders a localized version of the product page. |
+
+## Steps
+
+### Step 1: Path-based localization, subdomain-based localization, …?
+
+There are many ways to localize your store. This recipe focuses on path-based localization, but you could just as easily apply these concepts to subdomain-based localization, or header-based localization.
+
+Any of these approaches have different trade-offs, and the approach you take will depend on your project's requirements.
```

Comment:
Is it in scope of this recipe to list some of the pros/cons? Should we explain why it's best practice to express the market in the URL, instead of doing it in a cookie?